### PR TITLE
cmd/snap-bootstrap/initramfs-mounts: append uuid to ubuntu-data when decrypting

### PR DESCRIPTION
**Pull request by anonymouse64**
This will cause systemd-cryptsetup to mount ubuntu-data-enc at a mostly
unpredictable location in /dev/mapper.
Then requesting that we mount this node in /dev/mapper/ as the filesystem will
prevent an attacker from attaching a disk with an unencrypted partition named
ubuntu-data to the system and having snap-bootstrap mistakenly unlock the real
ubuntu-data-enc but mount the attacker's unencrypted ubuntu-data partition as
the real rootfs.

### CODE REVIEW

File: cmd/snap-bootstrap/cmd_initramfs_mounts.go [hunk 1]
```diff
@@ -612,13 +612,20 @@ func isDeviceEncrypted(name string) (ok bool, encdev string) {
 	return false, ""
 }
 
+// TODO:UC20: move this somewhere appropriate
+var readKernelUUID = func() string {
+	b, err := ioutil.ReadFile("/proc/sys/kernel/random/uuid")
+	if err != nil {
+		panic("can't read kernel uuid")
```
Comment (pedronis) [inline cmd/snap-bootstrap/cmd_initramfs_mounts.go:619]: nitpick: cannot ...

Comment (anonymouse64) [reply cmd/snap-bootstrap/cmd_initramfs_mounts.go:619]: will change in master, do you want this changed in 2.45 as well (with a followup or here)?

Comment (anonymouse64) [reply cmd/snap-bootstrap/cmd_initramfs_mounts.go:619]: fixed in master PR

File: cmd/snap-bootstrap/cmd_initramfs_mounts.go [hunk 2]
```diff
@@ -670,7 +679,17 @@ func unlockIfEncrypted(name string, lockKeysOnFinish bool) (string, error) {
 		return "", fmt.Errorf("cannot lock access to sealed keys: %v", lockErr)
 	}
 
-	return device, nil
+	// return the encrypted device if the device we are maybe unlocked is an
+	// encrypted device
```
Comment (xnox) [inline cmd/snap-bootstrap/cmd_initramfs_mounts.go:683]: nitpick: The comment sounds awkward. "have unlocked" ? and somehow change it, to not say "encrypted device" twice?
`// return the encrypted device if we attempted to unlock it` ?

Comment (anonymouse64) [reply cmd/snap-bootstrap/cmd_initramfs_mounts.go:683]: this was meant to read as 
```
return the encrypted device if the device we are maybe unlock**ing** is an encrypted device
```

Comment (anonymouse64) [reply cmd/snap-bootstrap/cmd_initramfs_mounts.go:683]: I'll change this in the master PR, not here

**Pull request merged**